core/port/pool_fixed.go

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package port
+
+// NewFixedPool creates new instance of PoolFixed
+func NewPoolFixed(port Port) *PoolFixed {
+	return &PoolFixed{
+		port:       port,
+		randomPool: NewPool(),
+	}
+}
+
+// PoolFixed hands out a fixed port for service use
+type PoolFixed struct {
+	port       Port
+	randomPool *Pool
```

Comment:
Whats the point to have Fixed pool, when its giving out random ports? 
Do we really need to create separate structure each time we need additional method?
Basically we duplicate code and scatter code with lots of instances of pools. Changes would mean we need to change them all.

Reply:
I'll also add a question:

How do you solve the possible collisions between the two different instances of port pools in a case when the provider is running both wireguard and openvpn?

Reply:
Collision wasn't solved, I can works towards that problems next.
Basically I am making more more clear signature of port acquire, because now is coupled and sensitive on changes

Reply:
But you're making it worse - there was no collision before, you're introducing it now.

Reply:
I checked:
- Collisions protection is there by trying to open socket on aquired ports
- Before acquired ports was cached to different variables `port` and `openvpnPort`, so it's the same